File: go/problems/day10_test.go
package problems

import "testing"

var day10 Solvable[int] = Day10{}

func TestDay10_TestSet_ShouldYield_Result(t *testing.T) {
	tests := map[string]int{
		"day10_1": 4,
		"day10_2": 8,
	}

	for problem, expected := range tests {
		input := readProblemInput(problem)
		result := day10.Solve(input)

		if result != expected {
			t.Errorf("expected %d but got %d", expected, result)
		}
	}
}

func TestDay10_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day10_3")
	result := day10.Solve(input)

	var expected int = 6867
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day10.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day10.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 10", () => {
  test("test set 1 should yield expected result", () => {
    const input = readProblemInput("day10_1");
    const result = solve(input);
    assert.equal(result, 4);
  });

  test("test set 2 should yield expected result", () => {
    const input = readProblemInput("day10_2");
    const result = solve(input);
    assert.equal(result, 8);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day10_3");
    const result = solve(input);
    assert.equal(result, 6867);
  });
});
